service/src/main/resources/api/service_openapi.yaml

```diff
@@ -1646,20 +1648,25 @@ components:
 
     HintQuery:
       type: object
-      description: Get display hints
+      description: |
+        Get display hints. Only one of relatedEntity or filter may be provided. 
+        Use the filter field to generate hints for the entity based on a filter.
       properties:
         relatedEntity:
           description: |
             Optional related entity.
-            If specified, then we return hints computed across related instances (e.g. condition occurrences where condition id=11).
-            Otherwise, we return hints computed across all instances (e.g. all persons).
+            If specified, then we return hints computed across related instances 
+            (e.g. condition occurrences where condition id=11). Otherwise, 
+            we return hints computed across all instances (e.g. all persons).
           type: object
           properties:
             name:
               type: string
               description: Related entity name
             id:
               $ref: "#/components/schemas/Literal"
+        filter:
```

Comment:
@tjennison-work query had a filter field as well. Extra fields like return attributes are skipped by using this